botocore/regions.py

```diff
@@ -661,59 +661,113 @@ def auth_schemes_to_signing_context(self, auth_schemes):
         if self._requested_auth_scheme == UNSIGNED:
             return 'none', {}
 
-        # normalize auth type names by removing any "sig" prefix
-        def strip_sig(auth_name):
-            return auth_name[3:] if auth_name.startswith('sig') else auth_name
-
         auth_schemes = [
-            {**el, 'name': strip_sig(el['name'])} for el in auth_schemes
+            {**scheme, 'name': self._strip_sig_prefix(scheme['name'])}
+            for scheme in auth_schemes
         ]
-
-        for scheme in auth_schemes:
-            if scheme['name'] not in AUTH_TYPE_MAPS:
-                continue
-
-            signing_context = {}
-            if 'signingRegion' in scheme:
-                signing_context['region'] = scheme['signingRegion']
-            elif 'signingRegionSet' in scheme:
-                # todo: can we handle lists of regions?
-                if len(scheme['signingRegionSet'][0]) > 0:
-                    signing_context['region'] = scheme['signingRegionSet'][0]
-            if 'signingName' in scheme:
-                signing_context.update(signing_name=scheme['signingName'])
-            if 'disableDoubleEncoding' in scheme:
-                signing_context['disableDoubleEncoding'] = ensure_boolean(
-                    scheme['disableDoubleEncoding']
+        if self._requested_auth_scheme is not None:
+            try:
+                # Use the first scheme that matches the requested scheme,
+                # after accounting for naming differences between botocore and
+                # endpoint rulesets. Keep the requested name.
+                name, scheme = next(
+                    (self._requested_auth_scheme, s)
+                    for s in auth_schemes
+                    if self._does_botocore_authname_match_ruleset_authname(
+                        self._requested_auth_scheme, s['name']
+                    )
                 )
-
-            LOG.debug(
-                'Selected auth type "%s" with signing context params: %s',
-                scheme['name'],
-                signing_context,
-            )
-            return scheme['name'], signing_context
+            except StopIteration:
+                # For legacy signers, no match will be found. Do not raise an
+                # exception, instead default to the logic in botocore
+                # customizations.
+                return None, {}
```

Comment:
I wonder since we do expect some cases to not match if this would be better written as a for loop?
```suggestion
for scheme in auth_schemes:
    if self._does_botocore_authname_match_ruleset_authname(
        self._requested_auth_scheme, scheme['name']
    ):
        name, scheme = self._requested_auth_scheme, scheme
        break

else:
    return None {}